.travis.yml

```diff
@@ -143,42 +168,36 @@ cache:
         - $HOME/.local
 
 install:
-    - test $TRAVIS_INSTALL_DEPS != On || ./scripts/install_deps.sh
+    - test $SOLC_INSTALL_DEPS_TRAVIS != On || ./scripts/install_deps.sh
     - test "$TRAVIS_OS_NAME" != "linux" || ./scripts/install_cmake.sh
     - echo -n "$TRAVIS_COMMIT" > commit_hash.txt
+    - test $SOLC_DOCKER != On || docker build -t ethereum/solc:$TRAVIS_COMMIT -f ./scripts/Dockerfile .
 before_script:
-    - test $TRAVIS_EMSCRIPTEN != On || ./scripts/build_emscripten.sh
-    - test $TRAVIS_RELEASE != On || (mkdir -p build
-      && cd build
-      && cmake .. -DCMAKE_BUILD_TYPE=$TRAVIS_BUILD_TYPE
-      && make -j2
-      && cd ..
-      && ./scripts/release.sh $ZIP_SUFFIX
-      && ./scripts/create_source_tarball.sh )
+    - test $SOLC_EMSCRIPTEN != On || ./scripts/build_emscripten.sh
+    - test $SOLC_RELEASE != On || ./scripts/build.sh $SOLC_BUILD_TYPE
 script:
-    - test $TRAVIS_DOCS != On || ./scripts/docs.sh
-
-    # There are a variety of reliability issues with the Solidity unit-tests at the time of
-    # writing (especially on macOS), so within TravisCI we will try to run the unit-tests
-    # up to 3 times before giving up and declaring the tests as broken.
-    #
-    # We should aim to remove this "retry logic" as soon as we can, because it is a
-    # band-aid for issues which need solving at their root.  Some of those issues will be
-    # in Solidity's RPC setup and some will be in 'eth'.  It seems unlikely that Solidity
-    # itself is broken from the failure messages which we are seeing.
-    #
-    # More details on known issues at https://github.com/ethereum/solidity/issues/769
-    - test $TRAVIS_TESTS != On || (cd $TRAVIS_BUILD_DIR && (./scripts/tests.sh || ./scripts/tests.sh || ./scripts/tests.sh) )
-env:
-    global:
-        - ENCRYPTION_LABEL="6d4541b72666"
-        - TRAVIS_BUILD_TYPE=RelWithDebInfo
-        - TRAVIS_DOCS=Off
-        - TRAVIS_EMSCRIPTEN=Off
-        - TRAVIS_INSTALL_DEPS=On
-        - TRAVIS_RELEASE=On
-        - TRAVIS_TESTS=On
+    - test $SOLC_DOCS != On || ./scripts/docs.sh
+    - test $SOLC_TESTS != On || (cd $TRAVIS_BUILD_DIR && ./scripts/tests.sh )
 
+# This is the deploy target for the dockerfile. If we are pushing into a develop branch, it will be tagged
+# as a nightly and appended the commit of the branch it was pushed in. If we are pushing to master it will
+# be tagged as "stable" and given the version tag as well.
+after_success:
+    - if [ "$TRAVIS_BRANCH" == "master" ]; then
```

Comment:
Oh right, this is a bit complicated. We want the docker image to be created only for the tag, but only if it is run on the release branch (by the way, it is "release" and not "master"). TRAVIS_BRANCH will be set to the tag name if it runs on a tag. I think the following should do: If TRAVIS_BRANCH starts with v and the prerelease.txt file is empty.